src/Group_User.php

```diff
@@ -682,6 +682,7 @@ public static function getRelationMassiveActionsSpecificities()
             'condition' => [
                 'is_usergroup' => 1,
             ] + getEntitiesRestrictCriteria(Group::getTable(), '', '', true)
+            + getEntitiesRestrictCriteria(User::getTable(), '', '', true)
```

Comment:
As far as I understand, this piece of code defines the dropdown options for the dropdown displayed in the `Associate to a group` and `Dissociate from a group` actions. The current filtering is probably OK on these actions.

Also, the `glpi_users.entities_id` field corresponds to the default user entity. Filtering the user must be done using the `Profile_User::getUserEntities()` result.

I did not take the time to try to reproduce the issue, but if there are a lack of rights checks here, it probably means that there are many things to check.

1. I think that `Group_User::canDeleteItem()` should be call by the massive action process. If the massive action permits to remove an item that is not supposed to be removed, then it is either not call by the massive action, either its result is not valid.
2. It seems that the `Add a user` feature in the `Users` tab of a group is also not correctly filtered, as it permit to associate users that should probably not be updatable in the current context.
3. The massive action on the user search page is not display when the user entities does not match the current session scope This is done by that : `{% elseif itemtype == 'User' and not can_view_all_entities() and not has_access_to_user_entities(row['id']) %}`.